server/runtime/src/main/java/org/infinispan/server/configuration/ServerConfigurationParser.java

```diff
@@ -1612,4 +1613,21 @@ private static void parseConnectorIpFilter(ConfigurationReader reader, IpFilterC
          }
       }
    }
+
+   @Override
+   public void readAttribute(ConfigurationReader reader, String elementName, int attributeIndex, ConfigurationBuilderHolder holder) {
+      if (org.infinispan.configuration.parsing.Element.forName(elementName) == org.infinispan.configuration.parsing.Element.TRANSPORT) {
+         ServerConfigurationBuilder serverBuilder = holder.getGlobalConfigurationBuilder().addModule(ServerConfigurationBuilder.class);
+         String attributeName = reader.getAttributeName(attributeIndex);
+         switch (Attribute.forName(attributeName)) {
+            case SECURITY_REALM:
+               serverBuilder.security().transport().securityRealm(reader.getAttributeValue(attributeIndex));
+               break;
+            default:
+               throw ParseUtils.unexpectedAttribute(reader, attributeName);
+         }
+      } else {
+         throw ParseUtils.unexpectedAttribute(reader, attributeIndex);
```

Comment:
should be element and not attribute.
```suggestion
         throw ParseUtils.unexpectedElement(reader, org.infinispan.configuration.parsing.Element.forName(elementName));
```

Reply:
Done